src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,32 @@
+package task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+
+public class Deadline extends Task {
+
+    private LocalDate by;
+
+    public Deadline(String name, String by) throws DukeException {
+        super(name);
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("incorrect date format");
+        }
+
+    }
+
+    @Override
+    public String saveTask() {
+        return "deadline|" + this.getName() + "/by" + by + (this.isDone() ? "|1" : "|0");
```

Comment:
This line is also a bit too long. Maybe we can write
    return "deadline |" + this.getName() + "/by" + by
            + (this.isDone() ? "|1"  : "|0");